src/averageAges.js

```diff
@@ -15,11 +15,28 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let peopleNew = [];
+  let count = 0;
+
+  function callback(arr) {
+    return (Math.ceil(arr.died / 100) === century);
+  };
+
+  if (century !== undefined) {
+    peopleNew = people.filter(callback);
+  } else {
+    peopleNew = people;
+  }
+
+  peopleNew = peopleNew.filter((arr) => (arr.sex === 'm'));
+
+  function callbackSum(summa, n) {
+    return (summa + n.died - n.born);
+  }
+
+  count = peopleNew.reduce(callbackSum, 0);
+
+  return Number((count / peopleNew.length).toFixed(2));
```

Comment:
you can simplify it to:
```suggestion
  return peopleNew.reduce(callbackSum, 0) / peopleNew.length;
```